packages/bsky/src/api/app/bsky/notification/putActivitySubscription.ts

```diff
@@ -10,39 +11,60 @@ export default function (server: Server, ctx: AppContext) {
     auth: ctx.authVerifier.standard,
     handler: async ({ input, auth }) => {
       const actorDid = auth.credentials.iss
-      const {
-        subject,
-        key: existingKey,
-        activitySubscription: { post, reply },
-      } = input.body
-
+      const { subject, activitySubscription } = input.body
       if (actorDid === subject) {
         throw new InvalidRequestError('Cannot subscribe to own activity')
       }
 
-      const payload: SubjectActivitySubscription = {
-        subject,
-        activitySubscription: { post, reply },
-      }
-      const key = existingKey ?? TID.nextStr()
+      const existingKey = await getExistingKey(ctx, actorDid, subject)
+      const enabled = isActivitySubscriptionEnabled(activitySubscription)
+
       const stashInput = {
         actorDid,
         namespace: NamespaceAppBskyNotificationDefsSubjectActivitySubscription,
-        payload,
-        key,
+        payload: <SubjectActivitySubscription>{
+          subject,
+          activitySubscription,
+        },
+        key: existingKey ?? TID.nextStr(),
       }
 
       if (existingKey) {
-        await ctx.stashClient.update(stashInput)
+        if (enabled) {
+          await ctx.stashClient.update(stashInput)
+        } else {
+          await ctx.stashClient.delete(stashInput)
+        }
       } else {
-        await ctx.stashClient.create(stashInput)
+        if (enabled) {
+          await ctx.stashClient.create(stashInput)
+        } else {
+          throw new InvalidRequestError(
+            'Cannot create a disabled activity subscription',
+          )
```

Comment:
I would lean towards treating this as idempotent, especially since the whole thing works asynchronously.